File: backend/src/config/mpesa.config.ts
export const mpesaConfig = {
  env: process.env['MPESA_ENV'] || 'sandbox',
  consumerKey: process.env['MPESA_CONSUMER_KEY'],
  consumerSecret: process.env['MPESA_CONSUMER_SECRET'],
  shortcode: process.env['MPESA_SHORTCODE'] || '174379',
  passkey: process.env['MPESA_PASSKEY'],
  callbackUrl: process.env['MPESA_CALLBACK_URL'] || 'https://your-domain.com/api/webhooks/mpesa',
  timeoutUrl: process.env['MPESA_TIMEOUT_URL'] || 'https://your-domain.com/api/webhooks/mpesa/timeout',
  transactionType: process.env['MPESA_TRANSACTION_TYPE'] || 'CustomerPayBillOnline',
  initiatorName: process.env['MPESA_INITIATOR_NAME'] || 'testapi',
  securityCredential: process.env['MPESA_SECURITY_CREDENTIAL'],
  organizationShortcode: process.env['MPESA_ORGANIZATION_SHORTCODE'] || '174379',
  b2cShortcode: process.env['MPESA_B2C_SHORTCODE'],
};

export const getMpesaBaseUrl = () => {
  return mpesaConfig.env === 'production'
    ? 'https://api.safaricom.co.ke'
    : 'https://sandbox.safaricom.co.ke';
};

File: app/lib/core/config/mpesa_config.dart
import 'package:homepulse/core/config/constants.dart';

class MpesaConfig {
  MpesaConfig._();

  static String get shortcode => Constants.mpesaShortcode;
  static String get passkey => Constants.mpesaPasskey;
  static String get consumerKey => Constants.mpesaConsumerKey;
  static String get consumerSecret => Constants.mpesaConsumerSecret;
  static String get callbackUrl => Constants.mpesaCallbackUrl;
  static String get authUrl => 'https://sandbox.safaricom.co.ke/oauth/v1/generate?grant_type=client_credentials';
  static String get stkPushUrl => 'https://sandbox.safaricom.co.ke/mpesa/stkpush/v1/processrequest';
  static String get c2bUrl => 'https://sandbox.safaricom.co.ke/mpesa/c2b/v1/registerurl';
  static String get b2cUrl => 'https://sandbox.safaricom.co.ke/mpesa/b2c/v1/paymentrequest';
  static String get transactionStatusUrl => 'https://sandbox.safaricom.co.ke/mpesa/transactionstatus/v1/query';
  static String get accountBalanceUrl => 'https://sandbox.safaricom.co.ke/mpesa/accountbalance/v1/query';
  static int get stkPushTimeoutSeconds => 30;
  static String get transactionType => 'CustomerPayBillOnline';
  static String get partyA => '254700000000';
  static String get partyB => Constants.mpesaShortcode;
  static String get resultUrl => '${Constants.apiUrl}/mpesa/result';
  static String get queueUrl => '${Constants.apiUrl}/mpesa/queue';
  static bool get useSandbox => Constants.isDebug;
}
